🚧  Implement 2 to master (v0.2.0)

schettn opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently the `session.begin()` response does not include a anonymous status. #51 
- Currently the client does not apply with the newest engine version. #62


**What is the new behavior (if this is a feature change)?**
- Now the `session.begin()` response does include an anonymous status. Closes #51 
- Now the client applies with the newest engine release. #Closes 62


**Other information**:
- 🐍

schettn committed:
Add Config

A configuration config has been added
which currently includes the credentials
of the anonymous user.

schettn committed:
Exchange hard coded credentials with config credentials

By implementing the config, the credentials
can now be extracted from it.

schettn committed:
Update SnekSession Begin Return Value

Now the return includes whether
the authenticated user is anonymous
or not.

schettn committed:
Merge pull request #60 from snek-at/improve-tasks-aa1

Improve tasks aa1 to implement 2 (Update: Anonymous User)

schettn committed:
Update Registration Mutation

The mutation has been updated to
fit the currently newest engine version.

Ref: https://github.com/snek-at/engine/tree/sub

schettn committed:
Update GitlabServer Query

The query has been updated to
fit the currently newest engine version.

Ref: https://github.com/snek-at/engine/tree/sub

schettn committed:
Add AllUserPageUrls Query

The query has been added because the
AllPageUrl query which was used before
returned not only user pages.

Ref: https://github.com/snek-at/engine/tree/sub

schettn committed:
Update Profile Query

The query has been updated to
fit the currently newest engine version.
Now it uses a slug instead of URL.

Ref: https://github.com/snek-at/engine/tree/sub

schettn committed:
Update GitlabServerData Interface

The type declaration has been
improved.

schettn committed:
Update AllPageUrlData Interface

The type declaration has been
improved.

schettn committed:
Improve Code Quality

A missing whitespace has been
added by prettier.

schettn committed:
Add AllUserPageUrls Task and Interface

A task which uses the allUserPageUrls
query has been added.

schettn committed:
Fix Syntax Error in AllUserPageUrls

A syntax error has been fixed.

schettn committed:
Update Default SnekClient URL

The default API URL has been
updated due to a endpoint  notation
change in engine.snek.at.

schettn committed:
Update TokenAuth Mutation

Field which cannot longer be
queried are removed now.

schettn committed:
Update Whoami Query

Field which cannot longer be
queried are removed now.

schettn committed:
Update AllUserPageUrls Query

Url has been replaced by slug.
Also the title has been added,
which represents the plain username.

schettn committed:
Update Profile Task

Url has been replace with slug
due to changes in the profile query.

schettn committed:
Update Default SnekClient URL

The default API URL has been
updated due to CSRF protection
errors.

schettn committed:
Merge pull request #61 from snek-at/update-snekset-aa1

Update snekset aa1 to implement 2 (Migrate: Newest Engine Release 06.08.2020)

schettn committed:
Bump to v0.2.0

The version has been updated
to 0.2.0.

Aichnerc reviewed (approved):
LGTM

pinterid reviewed (approved):
LGTM

kleberbaum commented on src/config.json:
```diff
@@ -0,0 +1,6 @@
+    "username": "cisco",
+    "password": "ciscocisco"
+  }
+}
```
```suggestion
}

```

kleberbaum commented on src/config.json:
```diff
@@ -0,0 +1,6 @@
+    "username": "cisco",
+    "password": "ciscocisco"
+  }
+}
```
I think a YAML file would be 100 times cooler. 💯

schettn replied:
In contrast to YAML, JSON can be imported as follows:
```
//> Config
import Config from "../../../config.json";
```
This notation guarantees full IntelliSense.

---

A method which works with YAML and JSON is as follows:
```
var config = require("../../../config.json")
```

But this notation does not have any IntelliSense support without implementing addidional interfaces.

Due to the small scope of the configuration this change would results in more work than benefits.

kleberbaum commented on src/templates/snek/gql/mutations/user.ts:
```diff
@@ -17,17 +17,16 @@ import gql from "graphql-tag";
+    registration: registrationFormPage(
+      url: "/registration/",
+      token: $token,
+      values: $values) {
```
```suggestion
      values: $values
    ) {
```

kleberbaum reviewed (approved):
Should be ok to merge.

schettn committed:
Update registration mutation

It is updated due to engine
breaking changes.

schettn committed:
Update cache mutation

It is updated due to engine
breaking changes.

schettn committed:
Update gitlabserver query

It is updated due to engine
breaking changes.

schettn committed:
Update alluserpages query

It is updated due to engine
breaking changes.

schettn committed:
Update profile query

It is updated due to engine
breaking changes.

schettn committed:
Update profile task

The task is updated due to changes
in the profile query.

schettn committed:
Add type declaration

A return type has been added.

schettn committed:
Update token aliveness

Now a custom task checks and refreshes
the token.

schettn committed:
Merge pull request #65 from snek-at/improve-errorhandling-aa1

Improve errorhandling aa1 to implement 2

schettn committed:
Update profile query and task

Once again, the profile query and task have been
updated due to a structure change in engine.

schettn committed:
Improve profile task

The person task no longer accepts a slug but a
personName.

schettn committed:
Update cache mutation and task

The cacheing task requires a person name now.

schettn committed:
Merge pull request #66 from snek-at/update-snekset-ba1

Update snekset ba1 to implement 2 (Migrate: Latest Sub Version 01.09.2020)

schettn committed:
Add instagram session

A session which can handle the rest api access to
instagram/facebook has been implemented.

schettn committed:
Rename instagram uri to path

The uris has been renamed to paths.

schettn committed:
Optimise tasks and clients

- All tasks which are not directly required by a related session
 have been removed.
- The template classes have been removed completely.

schettn committed:
Merge branch 'add-instagram-aa1' into update-client-aa1

schettn committed:
Merge branch 'improve-templates-aa1' into update-client-aa1

schettn committed:
Support more HTTP methods

A new fetch method has been implemented
to support more HTTP methods like
GET, POST, PUT, PATCH and DELETE.

schettn committed:
Merge branch 'improve-scraper-aa2' into update-client-aa1

schettn committed:
Fix invalid up to date token

The method was not correctly awaited and therefore returned
a promise instead of a token.
Also the handleErrors method lead to a loop, this has been fixed
now by only calling this.run() when a token error occurs.

schettn committed:
Merge branch 'improve-templates-ba1' into update-client-aa1

schettn committed:
Improve snek session begin return type

The return type of the snek session begin method has
been improved.

schettn committed:
Improve instagram client defaults

The default values and documentation of the instagram
client have been improved.

schettn committed:
Fix task run error loop

The loop which was created when a error had occurred
has been fixed now.

schettn committed:
Add cookie options

Secure and sameSite option has been added.

schettn committed:
Add support for uploading files

Now it is possible to upload files via the Apollo-Client.
Also Apollo v3 is implemented now.

schettn committed:
Fix token refresh issue with invalid refresh token

Now it is possible to refresh a session even when the refresh
token is invalid.

schettn committed:
Cleanup

Fix spacing, remove console logs and legacy error log.

schettn committed:
Update package.json

Remove several unnecessary packages.

schettn committed:
Implement codacy suggestions

Now handleErrors return null instead of undefined.
Also a missing equal sign has been added.

schettn committed:
Apply suggestions from code review

Co-authored-by: Pinterics David <55298934+pinterid@users.noreply.github.com>

schettn committed:
Improve regions and comments

The regions and comments have been improved.

schettn committed:
Apply suggestions from code review

Co-authored-by: Florian Kleber <kleberbaum@erebos.xyz>

schettn committed:
Improve comments

An extremely long region has been decimated.

schettn committed:
Merge pull request #67 from snek-at/update-client-aa1

Update client aa1 to implement 2

schettn committed:
Hotfix: Change compiler to babel

The compiler had to be changed in order to
support optional chaining etc.

schettn committed:
Add support for NodeJS

Now cookies are only used if the package is used
in browser.
